src/openklant/components/klantinteracties/api/serializers/internetaken.py

```diff
@@ -68,20 +71,43 @@ class Meta:
             },
         }
 
+    def validate(self, attrs):
+        status = attrs.get("status", None)
+        if not status and self.instance:
+            status = self.instance.status
+
+        if attrs.get("afgehandeld_op") and status == Taakstatus.te_verwerken.value:
+            raise serializers.ValidationError(
+                {
+                    "afgehandeld_op": _(
+                        "De Internetaak kan geen afgehandeld op datum bevatten "
+                        "als de status nog in '{te_verwerken}' staat."
+                    ).format(te_verwerken=Taakstatus.te_verwerken.value)
+                }
+            )
+
+        return super().validate(attrs)
+
     @transaction.atomic
     def create(self, validated_data):
         actor_uuid = str(validated_data.pop("actor").get("uuid"))
         klantcontact_uuid = str(validated_data.pop("klantcontact").get("uuid"))
+        afgehandeld_op = validated_data.pop("afgehandeld_op", timezone.now())
 
         validated_data["actor"] = Actor.objects.get(uuid=actor_uuid)
         validated_data["klantcontact"] = Klantcontact.objects.get(
             uuid=klantcontact_uuid
         )
 
+        if validated_data.get("status") == Taakstatus.verwerkt:
+            validated_data["afgehandeld_op"] = afgehandeld_op
```

Comment:
Is it necessary to implement this in the serializer `.create` and `.update`? I think it would be easier to implement it in the `InterneTaak` model `.save`.

I think since `afgehandeldOp` is a field for which the value is always `now()` or `None`, it should probably be a read-only field on the serializer, that way you can remove the `.validate` and the changes in `.create` and `.update` and leave the logic in `InterneTaak.save` as is

Reply:
Voor `afgehandeldOp` we want the user to either provide a value or user the date when the request was made. Hence why we shouldn't set the field as read_only. We can definitely set the value if none provided in the model `.save` method